### ISSUE TITLE
mllama patch modifies nn.LayerNorm globally

### ISSUE THREAD

**Issue by tyler-romero** [linkedin/Liger-Kernel#315]
### 🐛 Describe the bug

Instead of only patching the transformers mllama module (`transformers.models.mllama.modeling_mllama`), `apply_liger_kernel_to_mllama` modifies `torch.nn.LayerNorm` globally.

The issue is [here](https://github.com/linkedin/Liger-Kernel/commit/6ab3b9febc29f5045e6d2e27ba6bacaa4f041d91#diff-376c16dd1328612cf488158c6ce9805b044773659c8459cdcb2c6ec35dac346bR163).

The fix would be to:
(1) Not patch LayerNorm in Liger by assigning to `modeling_mllama.nn.LayerNorm`
(2) Change `transformers.models.mllama.modeling_mllama` to not use `from torch import nn` and to instead just import layernorm like `from torch.nn import LayerNorm`
(3) instead patch layernorm in Liger by assigning to `modeling_mllama.LayerNorm`

### Reproduce

```bash
pip install transformers==4.45 liger-kernel-nightly
```
```python
from liger_kernel.transformers import apply_liger_kernel_to_mllama
from torch import nn

apply_liger_kernel_to_mllama()
print(nn.LayerNorm)
<class 'liger_kernel.transformers.layer_norm.LigerLayerNorm'>
```

### Versions

Environment Report:
-------------------
Operating System: Linux-6.1.85+-x86_64-with-glibc2.35
Python version: 3.10.12
PyTorch version: 2.4.1+cu121
CUDA version: Not available
Triton version: 3.1.0
Transformers version: 4.45.0

**Comment by ByronHsu** [comment]
curious does it require to change transformer source code? i think we can maybe raise a request

**Comment by tyler-romero** [comment]
Yeah the proposed fix would require a change to `transformers` unfortunately. How mllama was implemented differs very slightly from the conventions in other transformers modeling files.

**Comment by ByronHsu** [comment]
sounds good. let us try to send a PR there
